packages/appeals-service-api/src/routes/v2/migrate/controller.js

```diff
@@ -0,0 +1,197 @@
+		await processAppeal(doc);
+	}
+
+	res.status(200).send(result);
+
+	/**
+	 * @returns {Promise<[string]>}
+	 */
+	async function getExistingIds() {
+		const existingAppealsResult = await sqlClient.appeal.findMany({
+			select: {
+				legacyAppealSubmissionId: true
+			},
+			where: {
+				NOT: [
+					{
+						legacyAppealSubmissionId: null
+					}
+				]
+			}
+		});
+
+		return existingAppealsResult.map((obj) => obj.legacyAppealSubmissionId);
+	}
+
+	/**
+	 * @returns {Promise<Map<string, string>>}
+	 */
+	async function getExistingUsersMap() {
+		const existingUsersResult = await sqlClient.appealUser.findMany({
+			select: {
+				email: true,
+				id: true
+			}
+		});
+
+		return new Map(existingUsersResult.map((obj) => [obj.email, obj.id]));
+	}
+
+	/**
+	 * @param {[string]} [existingIds]
+	 * @returns {Promise<import('mongodb').Cursor>}
+	 */
+	async function findAppeals(existingIds) {
+		let lookupQuery = {};
+
+		if (existingIds) {
+			lookupQuery = {
+				_id: { $nin: existingIds }
+			};
+		}
+
+		return await mongodb
+			.get()
+			.collection('appeals')
+			.find(lookupQuery, {
+				batchSize: BATCH_SIZE,
+				projection: {
+					_id: 1,
+					'appeal.email': 1,
+					'appeal.decisionDate': 1,
+					'appeal.state': 1
+				}
+			});
+	}
+
+	/**
+	 * @typedef {Object} AppealQueryResult
+	 * @property {string} _id
+	 * @property {{email: string, decisionDate: string, state: string}} appeal
+	 */
+
+	/**
+	 * @param {AppealQueryResult} doc
+	 * @returns {Promise<void>}
+	 */
+	async function processAppeal(doc) {
+		try {
+			result.total++;
+
+			// check if appeal already processed
+			if (existingIds.includes(doc._id)) {
+				result.alreadyProcessed++;
+				return;
+			}
+
+			// skip appeals without an required data
+			if (!doc.appeal.email || !doc.appeal.decisionDate || !doc.appeal.state) {
+				result.skipped++;
+				return;
+			}
+
+			// add to sql
+			const userId = await getOrCreateUser(doc.appeal.email);
+			await createAndLinkAppeal(doc, userId);
+
+			result.migrated++;
+		} catch (err) {
+			result.errors.push({ appealId: doc._id, error: err.message });
+		}
+	}
+
+	/**
+	 * @param {string} email
+	 * @returns {Promise<string>}
+	 */
+	async function getOrCreateUser(email) {
+		let userId = existingUserMap.get(email);
+
+		if (!userId) {
+			const user = await sqlClient.appealUser.create({
+				data: {
+					email: email
+				}
+			});
+
+			existingUserMap.set(email, user.id);
+			userId = user.id;
+		}
+
+		return userId;
+	}
+
+	/**
+	 * @param {AppealQueryResult} doc
+	 * @param {string} userId
+	 */
+	async function createAndLinkAppeal(doc, userId) {
+		const appeal = await sqlClient.appeal.create({
+			data: {
+				legacyAppealSubmissionId: doc._id,
+				legacyAppealSubmissionDecisionDate: doc.appeal.decisionDate,
+				legacyAppealSubmissionState: doc.appeal.state
+			}
+		});
+
+		await sqlClient.appealToUser.create({
+			data: {
+				appealId: appeal.id,
+				userId: userId,
+				role: 'appellant'
```

Comment:
Do we not need to set this from the data? could be agent?

Reply:
ah yes, will look into that

Reply:
Have resolved that and wrapped in transaction
Have also added a few other checks/options a separate commit